examples/corda_finetuning/README.md

```diff
@@ -0,0 +1,252 @@
+# CorDA: Context-Oriented Decomposition Adaptation of Large Language Models for Task-Aware Parameter-Efficient Fine-tuning
+
+## Introduction
+
+
+Existing PEFT methods are mostly agnostic of the context of a task of concern, e.g., a downstream task to learn or some pre-trained world knowledge to maintain.
+[CorDA](https://openreview.net/pdf?id=Gi00NVru6n) builds task-aware LoRA adapters from weight decomposition oriented by the context of the task concerned. 
+
+Concretely, CorDA randomly collects a few (by default 256) data samples from a target task, e.g. questions from a QA dataset or instructions to write a code or solve a math problem, and feeds these samples into a pre-trained LLM. We can obtain the covariance matrix of the input activation of each linear layer, i.e., $C=XX^T\in\mathcal{R}^{d_{in}\times d_{in}}$. 
```

Comment:
> (by default 256)

This is no longer true, is it?

Reply:
It is still 256 in our provided example, see [this argument](https://github.com/huggingface/peft/blob/5a06739573281bd2e4586cbed1ebe35f6ca12d61/examples/corda_finetuning/preprocess.py#L111). For user's own preprocessing, it will be the size of the dataloader provided by users. I have changed "by default 256" into "by default 256 in our `preprocess.py`".

Reply:
`calib_loader_size` determines the size of the calibration dataset (sample count) in our `preprocess.py`. 

For users, they provide the calibration dataloader themselves, and pass it to  `run_model` in [this line ](https://github.com/huggingface/peft/blob/d5f0a44fecfeb1df6cec425b34877dfab39a1e66/examples/corda_finetuning/preprocess.py#L73). In our previous version, the users need to specify `sample_count` as an argument in `CordaConfig` and need to ensure it equals to the size of the dataloader they provide. So we remove this argument in [this update](https://github.com/huggingface/peft/pull/2231/commits/5a06739573281bd2e4586cbed1ebe35f6ca12d61), and calculate `sample_count` automatically in [this line](https://github.com/huggingface/peft/blob/d5f0a44fecfeb1df6cec425b34877dfab39a1e66/src/peft/tuners/lora/corda.py#L182C1-L182C9). As a result, uses only need to pass a dataloader with a determined `calib_loader_size `, and do not need to specify the same sample count again in CordaConfig.